pkg/reconciler/app/resources/source.go

```diff
@@ -16,28 +16,50 @@ package resources
 
 import (
 	"fmt"
+	"path"
+
+	"strconv"
 
 	"github.com/google/kf/pkg/apis/kf/v1alpha1"
 	"github.com/knative/serving/pkg/resources"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"knative.dev/pkg/kmeta"
 )
 
+const buildComponentName = "build"
+
 // MakeSourceLabels creates labels that can be used to tie a source to a build.
 func MakeSourceLabels(app *v1alpha1.App) map[string]string {
-	return app.ComponentLabels("build")
+	return app.ComponentLabels(buildComponentName)
+}
+
+// BuildpackBulidImageDestination gets the image name for an application build.
+func BuildpackBulidImageDestination(app *v1alpha1.App, space *v1alpha1.Space, suffix int64) string {
+	registry := space.Spec.BuildpackBuild.ContainerRegistry
+	image := fmt.Sprintf("app-%s-%s:%s", app.Namespace, app.Name, strconv.FormatInt(suffix, 36))
+
+	return path.Join(registry, image)
 }
 
 // MakeSource creates a source for the given application.
-func MakeSource(app *v1alpha1.App, space *v1alpha1.Space) (*v1alpha1.Source, error) {
+//
+// Suffix must be a unique base-36 string suitable for using as a label or name.
```

Comment:
nit: This function comment doesn't look right. Should start with MakeSource, and suffix is an int64, not a string.